src/wasm/wasm-s-parser.cpp

```diff
@@ -2824,19 +2824,46 @@ void SExpressionWasmBuilder::parseTable(Element& s, bool preParseImport) {
 void SExpressionWasmBuilder::parseElem(Element& s) {
   Index i = 1;
   Table* table = nullptr;
+  Expression* offset = nullptr;
 
   if (!s[i]->isList()) {
-    auto name = getTableName(*s[i]);
-    table = wasm.getTableOrNull(name);
+    // optional segment id OR 'declare' OR start of elemList
+    i += 1;
+  }
+
+  bool oldStyle = true;
```

Comment:
It would be good to have a comment with examples of the old style and the new style.

Reply:
Done.